src/NCronJob/Registry/JobRun.cs

```diff
@@ -7,19 +7,24 @@ internal class JobRun
 {
     private readonly JobRun rootJob;
     private int jobExecutionCount;
+    private readonly TimeProvider timeProvider;
     private readonly Action<JobRun> progressReporter;
     private readonly ConcurrentBag<JobRun> pendingDependents = [];
 
     private JobRun(
+        TimeProvider timeProvider,
         JobDefinition jobDefinition,
+        DateTimeOffset runAt,
```

Comment:
Does it make sense to have the `TImeProvider` and `runAt`?
Given that `runat` will always be resolved to `GetUtcNow()`? 

Therefore: Can we remove `runAt` as parameter?

Reply:
The alternative would be somewhat the design as before:
`JobRun` doesn't know anything about the `TimeProvider` but gets passed in all the necessary `DateTimeOffset` objects.

So `IsExpired` would ask for a `DateTimeOffset` as a parameter rather than the `TimeProvider` itself. What do you think of this design?

Reply:
It's been done this way to account for the `InstantRegistry` that allows to schedule a job for later execution.
As such, `runAt` may actually be different from UtcNow

Reply:
Okay - that makes sense. Then we should go down the path you did!

Reply:
I've just pushed another proposal. The code has been tweaked to reduce the diff noise and account for this `runAt` parameter only for the use case of the InstantRegistry.

Does it read better?

Reply:
Lol :D without the direct comparison I can not even tell you what really changed :D
In any case I like it.